common/src/stack/command/stack/commands/load/json/plugin_host.py

```diff
@@ -0,0 +1,352 @@
+						self.owner.successes += 1
+					if interface['name']:
+						self.owner.command('set.host.interface.name', [
+											host_name,
+											f'name={interface["name"]}',
+											f'interface={interface["interface"]}'
+						])
+						self.owner.log.info(f'success setting {host["name"]} interface name')
+						self.owner.successes += 1
+					if interface['module']:
+						self.owner.command('set.host.interface.module', [
+											host_name,
+											f'module={interface["module"]}',
+											f'interface={interface["interface"]}'
+						])
+						self.owner.log.info(f'success setting {host["name"]} interface module')
+						self.owner.successes += 1
+					if interface['vlan']:
+						self.owner.command('set.host.interface.vlan', [
+											host_name,
+											f'vlan={interface["vlan"]}',
+											f'interface={interface["interface"]}'
+						])
+						self.owner.log.info(f'success setting {host["name"]} interface vlan')
+						self.owner.successes += 1
+					if interface['options']:
+						self.owner.command('set.host.interface.options', [
+												host_name,
+												f'options={interface["options"]}',
+												f'interface={interface["interface"]}'
+						])
+						self.owner.log.info(f'success setting {host["name"]} interface options')
+						self.owner.successes += 1
+					if interface['channel']:
+						self.owner.command('set.host.interface.channel', [
+												host_name,
+												f'channel={interface["channel"]}',
+												f'interface={interface["interface"]}'
+						])
+						self.owner.log.info(f'success setting {host["name"]} interface channel')
+						self.owner.successes += 1
+
+
+				except CommandError as e:
+					if 'exists' in str(e):
+						self.owner.log.info(f'warning setting host interface {interface["name"]}: {e}')
+						self.owner.warnings += 1
+					else:
+						self.owner.log.info(f'error setting host interface {interface["name"]}: {e}')
+						self.owner.errors += 1
+
+
+
+			# iterate through each attr for the host and add it
+			for attr in host['attrs']:
+				attr_name = attr['name']
+				attr_value = attr['value']
+				if not isinstance(attr_value, str):
+					attr_value = ' '.join(attr['value'])
+				attr_shadow = attr['shadow']
+				try:
+					self.owner.command('set.host.attr', [
+									host_name,
+									f'attr={attr_name}',
+									f'value={attr_value}',
+									f'shadow={attr_shadow}'
+					])
+					self.owner.log.info(f'success setting {host["name"]} attr {attr_name}')
+					self.owner.successes += 1
+
+				except CommandError as e:
+					if 'exists' in str(e):
+						self.owner.log.info(f'warning setting {host["name"]} attr {attr_name}: {e}')
+						self.owner.warnings += 1
+					else:
+						self.owner.log.info(f'error setting {host["name"]} attr {attr_name}: {e}')
+						self.owner.errors += 1
+
+			#iterate through each firewall rule and add it
+			for rule in host['firewall']:
+				try:
+					self.owner.command('add.host.firewall', [
```

Comment:
Are some of these parameters optional?  Does add firewall care if they're passed as `''`?